pkg/kubelet/nodeconfig/verify.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeconfig
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"regexp"
+	"sort"
+
+	apiv1 "k8s.io/kubernetes/pkg/api/v1"
+)
+
+const (
+	sha256Alg = "sha256"
+)
+
+// verifiable is for verifying the integrity of a checkpointed configuration (mainly intended to guard against API object mutation)
+type verifiable interface {
+	verify() (parsable, error)
+}
+
+// verifiableConfigMap is for verifying integrity of config from ConfigMaps
+type verifiableConfigMap struct {
+	cm *apiv1.ConfigMap
+}
+
+// verify checks that re-hashing the data in the ConfigMap results in the hash in the ConfigMap's name.
+// If verification fails, returns an error.
+// If verification succeeds, returns a parsable. This interface can be used to parse the ConfigMap into a KubeletConfiguration.
+func (v *verifiableConfigMap) verify() (parsable, error) {
+	name, alg, expectedHash, err := parseConfigName(v.cm.ObjectMeta.Name)
```

Comment:
so changing a config means creating a new configmap and updating all nodes to point to the new config map?